### ISSUE TITLE
iptables display wrong rules

### ISSUE THREAD

**Issue by ptpt52** [openwrt/openwrt#11169]
setup rules by:
```
iptables -t mangle -A PREROUTING -s 192.168.20.80/29  -m mark --mark 0x0/0xff000000 -m conntrack --ctstate NEW -m comment --comment natcap-rule -j MARK --set-xmark 0xd000000/0xff000000
```

iptables-save: display wrong output:
```
# Generated by iptables-save v1.8.8 (nf_tables) on Sat Nov  5 16:36:58 2022
*mangle
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
-A PREROUTING -s 0.0.0.0/32 -m mark --mark 0x0/0xff000000 -m conntrack --ctstate NEW -m comment --comment natcap-rule -j MARK --set-xmark 0xd000000/0xff000000
COMMIT
# Completed on Sat Nov  5 16:36:58 2022
```

the display shows '-s 0.0.0.0/32' but it should be "-s 192.168.20.80/29"

**Comment by ptpt52** [comment]
installed ipks:
```
iptables-mod-conntrack-extra - 1.8.8-1
iptables-mod-ipopt - 1.8.8-1
iptables-nft - 1.8.8-1
```

no  iptables-legacy

**Comment by ptpt52** [comment]
this ping @hauke @champtar

**Comment by champtar** [comment]
I can reproduce it with Fedora 37
-> reported upstream https://bugzilla.netfilter.org/show_bug.cgi?id=1642

**Comment by ptpt52** [comment]
> I can reproduce it with Fedora 37 -> reported upstream https://bugzilla.netfilter.org/show_bug.cgi?id=1642

looks like fixed by commit f315af1cf8871 ("nft: track each register individually")
could you help backport it? @champtar

**Comment by Spudz76** [comment]
I modded the package to pull the latest upstream git master from last night (`39589b3edb`) and it has this and a bunch of other fixes (but could have even more, different bugs of course).  But, it builds...

[iptables-upstream-git-master-39589b3edb.patch.txt](https://github.com/openwrt/openwrt/files/10144361/iptables-upstream-git-master-39589b3edb.patch.txt)

**Comment by Neustradamus** [comment]
To follow this ticket

**Comment by rodsmar** [comment]
I submitted an PR to backport this commit.

> > I can reproduce it with Fedora 37 -> reported upstream https://bugzilla.netfilter.org/show_bug.cgi?id=1642
> 
> looks like fixed by commit f315af1cf8871 ("nft: track each register individually") could you help backport it? @champtar
